plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/interceptor/ProducerSendInterceptor.java

```diff
@@ -0,0 +1,306 @@
+            doInAfterTrace(recorder, target, args, result, throwable);
+        } catch (Throwable th) {
+            if (logger.isWarnEnabled()) {
+                logger.warn("AFTER. Caused:{}", th.getMessage(), th);
+            }
+        } finally {
+            trace.traceBlockEnd();
+            if (isAsyncTraceDestination(trace)) {
+                deleteTrace(trace);
+            }
+        }
+    }
+
+    protected void doInBeforeTrace(SpanEventRecorder recorder, Object target, Object[] args) {
+        recorder.recordServiceType(RocketMQConstants.ROCKETMQ_CLIENT);
+        recorder.recordApi(methodDescriptor);
+
+        final String endPoint = ((EndPointFieldAccessor) target)._$PINPOINT$_getEndPoint();
+        recorder.recordEndPoint(endPoint);
+        recorder.recordDestinationId(endPoint);
+
+        final SendMessageRequestHeader sendMessageRequestHeader = (SendMessageRequestHeader) args[3];
+        recorder.recordAttribute(RocketMQConstants.ROCKETMQ_TOPIC_ANNOTATION_KEY,
+                                 sendMessageRequestHeader.getTopic());
+        recorder.recordAttribute(RocketMQConstants.ROCKETMQ_PARTITION_ANNOTATION_KEY,
+                                 sendMessageRequestHeader.getQueueId());
+        final Trace trace = traceContext.currentRawTraceObject();
+        final TraceId nextTraceId = trace.getTraceId().getNextTraceId();
+        recorder.recordNextSpanId(nextTraceId.getSpanId());
+        // set header
+        final Map<String, Object> paramMap = new HashMap<>();
+        if (trace.canSampled()) {
+            final StringBuilder properties = new StringBuilder(sendMessageRequestHeader.getProperties());
+            paramMap.put(Header.HTTP_FLAGS.toString(), String.valueOf(nextTraceId.getFlags()));
+            paramMap.put(Header.HTTP_PARENT_APPLICATION_NAME.toString(), traceContext.getApplicationName());
+            paramMap.put(Header.HTTP_PARENT_APPLICATION_TYPE.toString(),
+                         String.valueOf(traceContext.getServerTypeCode()));
+            paramMap.put(Header.HTTP_PARENT_SPAN_ID.toString(), String.valueOf(nextTraceId.getParentSpanId()));
+            paramMap.put(Header.HTTP_SPAN_ID.toString(), String.valueOf(nextTraceId.getSpanId()));
+            paramMap.put(Header.HTTP_TRACE_ID.toString(), nextTraceId.getTransactionId());
+            paramMap.put(RocketMQConstants.ACCEPTOR_HOST, endPoint);
+            paramMap.put(RocketMQConstants.IS_ASYNC_SEND, trace.isAsync());
+
+            for (Map.Entry<String, Object> entry : paramMap.entrySet()) {
+                properties.append(entry.getKey());
+                properties.append(NAME_VALUE_SEPARATOR);
+                properties.append(entry.getValue());
+                properties.append(PROPERTY_SEPARATOR);
+            }
+            sendMessageRequestHeader.setProperties(properties.toString());
+        } else {
+            paramMap.put(Header.HTTP_SAMPLED.toString(), SAMPLING_RATE_FALSE);
```

Comment:
When you don't sample, it looks like you don't include properties in the header.
please check.

Reply:
> When you don't sample, it looks like you don't include properties in the header.
> please check.

Is there a need to pass properties when don't sample?

Reply:
Yes, it will be trigger that target server will be not sampled.